lib/transport/tls-context.c

```diff
@@ -1003,6 +461,9 @@ tls_context_setup_context(TLSContext *self)
   if (!self->ssl_ctx)
     goto error;
 
+  if (!_setup_keylog_file(self))
```

Comment:
Hmm, something is weird here.
`_setup_keylog_file()` was called for each new sessions (connections), which opened a new file for appending, but it seems to be the same file.

Moving this here changes the time and frequency of opening this file.
The time does not seem to matter as this option is used for debugging.
The "frequency change" seems correct to me, so this probably fixed an fd leak too, right?

Reply:
It seems so. I misread one part of the code and I thought this was guarding against reopens, but now as I look at it again, this is not a guard, just checks whether the file name is set up.

```
  if(ctx->keylog_file_path)
    {
      if(!tls_session_keylog_setup_file(self, self->ctx->keylog_file_path))
        {
          tls_context_unref(self->ctx);
          g_free(self);
          return NULL;
        }
    }
```

So you are right, there has been an fd leak, but I didn't notice it.

But I think the new behaviour is a lot easier to follow.

We will open the file just once and will set the appropriate callbacks through SSL_CTX (and not SSL)

All connections will use the same file, in append mode, just as before.
I also tested this functionality manually.

Reply:
Thanks.